src/runtime/HalideBuffer.h

```diff
@@ -150,16 +150,17 @@ class Buffer {
         return halide_type_of<typename std::remove_cv<not_void_T>::type>();
     }
 
-    /** Is this Buffer responsible for managing its own memory */
-    bool manages_memory() const {
+    /** Does this Buffer own the host memory it refers to? */
+    bool owns_host_memory() const {
         return alloc != nullptr;
     }
 
 private:
     /** Increment the reference count of any owned allocation */
     void incref() const {
-        if (!manages_memory()) return;
-        alloc->ref_count++;
+        if (owns_host_memory()) {
+            alloc->ref_count++;
+        }
         if (buf.device) {
```

Comment:
For the example in the PR description, will this code path be exercised before `output_wrapped` is destructed? There is no copy or move ctor of Buffer called after the Halide pipeline allocates a device buffer internally. Maybe I missed something.